src/ray/common/task/task.cc

```diff
@@ -42,9 +42,7 @@ const std::string &RayTask::GetPreferredNodeID() const { return preferred_node_i
 void RayTask::ComputeDependencies() { dependencies_ = task_spec_.GetDependencies(); }
 
 std::string RayTask::DebugString() const {
-  std::ostringstream stream;
-  stream << "task_spec={" << task_spec_.DebugString() << "}";
-  return stream.str();
+  return absl::StrFormat("task_spec={%s}", task_spec_.DebugString());
```

Comment:
`std::ostringstream` is inefficient for formatting strings, since it stores the whole string in the internal buffer; when `str` is called, the whole buffer is **COPIED** out, instead of moved.

From C++20, we will have `std::move(ss).str()`.